python/tvm/relay/frontend/onnx.py

```diff
@@ -4898,6 +4898,90 @@ def _impl_v10(cls, inputs, attr, params):
         return out
 
 
+class QGemm(OnnxOpConverter):
+    """Operator converter for QGemm."""
+
+    @classmethod
+    def _impl_v1(cls, inputs, attr, params):
+        # http://www.xavierdupre.fr/app/mlprodict/helpsphinx/onnxops/onnx_commicrosoft_QGemm.html
+
+        a = inputs[0]
+        a_scale = get_scalar(inputs[1], params)
+        a_zp = get_scalar(inputs[2], params, "int32")
+
+        b = inputs[3]
+        #  a scalar or 1D tensor of size 1 or N
+        b_scale = get_scalar_or_1d_tensor(inputs[4], params)
+        #  a scalar or 1D tensor of size 1 or N
+        b_zp = get_scalar_or_1d_tensor(inputs[5], params, "int32")
+
+        # note that if optional and not provided then value will be None.
+        C = inputs[6] if len(inputs) > 6 else None
+        # must be null or a scalar or 1D tensor of size 1
+        y_scale = inputs[7] if len(inputs) > 7 else None
```

Comment:
If an optional input is not passed, then its default value is `None`. So you can simplify this block like this:
```Python
C = inputs[6]
y_scale = inputs[7]
```

Reply:
Done.